src/diem/offchain/client.py

```diff
@@ -54,12 +55,28 @@ class Client:
     def __post_init__(self) -> None:
         self.my_compliance_key_account_id = self.account_id(self.my_compliance_key_account_address)
 
-    def send_command(self, command: PaymentCommand, sign: typing.Callable[[bytes], bytes]) -> CommandResponseObject:
-        return self.send_request(
-            request_sender_address=command.my_actor_address,
-            opponent_account_id=command.opponent_actor_obj().address,
-            request_bytes=jws.serialize(command.new_request(), sign),
-        )
+    def send_command(self, command: Command, sign: typing.Callable[[bytes], bytes]) -> CommandResponseObject:
+        if command.command_type() == CommandType.PaymentCommand:
```

Comment:
command_type is same with type of the command, hence instead of adding command_type method, it is better to just do isinstance(command, PaymentCommand);
regardless type, it maybe better to add my_address() and opponent_address() to Command, then this method does not need check type.

process_inbound_request checks request.command_type, because it's protocol definition, ideally, should have isinstance check before typing.cast as validation of command_type matches command object type, because inbound data need to be carefully validated.

Reply:
@xli - I implemented the first part of your comment (my_address() and opponent_address() were added to Command so no changes was required in send_command), but I didn't understand the second part regarding "process_inbound_request"... please elaborate if you can, tx